src/tools/illink/test/ILLink.RoslynAnalyzer.Tests/TestCaseUtils.cs

```diff
@@ -24,11 +25,15 @@ public abstract class TestCaseUtils
 	{
 		private static readonly string MonoLinkerTestsCases = "Mono.Linker.Tests.Cases";
 
+		private static string targetFramework = (string)AppContext.GetData("ILLink.RoslynAnalyzer.Tests.TargetFramework")!;
+
+		private static string frameworkVersion = typeof(object).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()!.InformationalVersion.Split('+')[0];
+
 		public static readonly ReferenceAssemblies NetCoreAppReferencessemblies =
 		new ReferenceAssemblies (
-				"net8.0",
-				new PackageIdentity ("Microsoft.NETCore.App.Ref", "8.0.0"),
-				Path.Combine ("ref", "net8.0"))
+				targetFramework,
+				new PackageIdentity ("Microsoft.NETCore.App.Ref", frameworkVersion),
```

Comment:
```suggestion
				new PackageIdentity ("Microsoft.NETCore.App.Ref", $"net{Environment.Version.Major}.{Environment.Version.Minor}"),
```

Reply:
This one isn't right. This should be the package version to restore.

Reply:
Fixed with the latest commit.